File: playground/examples/node/hello.js
const { task, flow, runWorkflow } = require("sayiir");

const fetchUser = task("fetch-user", (id) => {
  return { id, name: "Alice" };
});

const sendEmail = task("send-email", (user) => {
  return `Sent welcome to ${user.name}`;
});

const workflow = flow("welcome")
  .then(fetchUser)
  .then(sendEmail)
  .build();

runWorkflow(workflow, 42).then(console.log);

File: playground/examples/python/hello.py
from sayiir import task, Flow, run_workflow

@task
def fetch_user(user_id: int) -> dict:
    return {"id": user_id, "name": "Alice"}

@task
def send_email(user: dict) -> str:
    return f"Sent welcome to {user['name']}"

workflow = Flow("welcome").then(fetch_user).then(send_email).build()
result = run_workflow(workflow, 42)
print(result)
